1st draft ready for review (#2361)
Kong/developer.konghq.com: app/_how-tos/export-an-api-spec-as-a-document.md

---

> title: Export your API design document or request data

Reviewer: ```suggestion
title: Export a design document in Insomnia
```
I think we need specificity since the paths for design documents and requests are different.

content_type: how_to

products:
  - insomnia

description: Export a design document, collection, or single request from Insomnia.

tags:
  - documents

tldr:
  q: How do I export an API design document, collection, or a request in Insomnia?
  a: In your workspace, open the Document or Collection menu, click **Preferences → Data**, select the target format and scope, and then export the file.

faqs:
  - q: Can I export just one request?

>     a: Yes. Click **Export Data → Single Request** from the Document or Collection menu or click  **Preferences → Data**. Then select just that request.

Reviewer: ```suggestion
    a: Select the collection drop-down menu, then **Export**, from here you can select an individual request.
```

Reviewer: This is what it looked like on my system: 
<img width="1197" height="658" alt="image" src="https://github.com/user-attachments/assets/eb1935f3-d557-4f7f-8ba6-b6b4e1452088" />

Reviewer: I think you can remove this FAQ item in favor of making the entire doc about exporting a design document. 

The requests information could be added in an existing reference doc.

prereqs:
  inline:

>     - title: API specification
>       content: |
>         You need to have an API specification in one of the following formats:
>           * Insomnia
>           * Postman v2
>           * HAR
>           * OpenAPI (versions 3.0, 3.1)
>           * Swagger
>           * WSDL
>           * cURL

Reviewer: ```suggestion
    - title: An Insomnia design document
      content: |
        You need to have an Insomnia design document. For instructions on creating one see [Create a design document](/how-to/create-a-design-document/)
```

Reviewer: Since the doc title is: "Create a design document" you only need a design document. I tested and it seems like you don't need anything in that design document to export it.

Reviewer: The original tl;dr you had was more relevant for importing, but it looks like once the information is imported it becomes either a document or a request collection.

      icon_url: /assets/icons/code.svg

related_resources:
  - text: API specs in Insomnia
    url: /insomnia/api-specs/
  - text: Design documents
    url: /insomnia/documents/
  - text: Import an API specification as a design document in Insomnia
    url: /how-to/import-an-api-spec-as-a-document/
  - text: Resource types reference
    url: /insomnia/resource-types-reference/   
  - text: Special resource IDs
    url: /insomnia/special-resource-ids/    
---

> Export your API design documents from Insomnia to capture the exact endpoints and schemas that you use. This ensures accurate documentation, seamless teamwork, and easy integration with tools like [Inso CLI](/inso-cli/) or CI/CD workflows.
> 

Reviewer: ```suggestion
```

You can export the following files from Insomnia:
- API design document
- Collection
- Request

> 

Reviewer: ```suggestion
This tutorial will explain how to export a document. 
```

## Export a file

To export a file from Insomnia, navigate to your workspace:
1. Click **Preferences**, and then click **Data**.

> 2. Select an export option:

Reviewer: ```suggestion
1. Select an export option:
```

Reviewer: It automatically renders as a numbered list.

>     - **Export the "name of file" Document**: Export a single, specific API design document.
>     - **Export the "project name" Project**: Export a project, including all associated API requests, design documents, and tests.
>     - **Export all data**: Export all data from your all your workspaces.
>     - **Create Run Button**: Create a sharable button that, when clicked, automatically opens a specific request or collection in the Insomnia application and executes it.

Reviewer: ```suggestion
    - **Export Document**: Export a single, specific API design document.
    - **Export Project**: Export a project, including all associated API requests, design documents, and tests.
    - **Export all data**: Export all data from your all your workspaces.
    - **Create Run Button**: Create a sharable button that, when clicked, automatically opens a specific request or collection in the Insomnia application and executes it.
```

3. Click **Export**.
4. Choose a format to export the file as:
    - **Insomnia v5**: A native JSON export format that contains the full structure of Insomnia projects, including requests, environments, metadata, and documentation.
    - **HAR - HTTP Archive Format**: A standardized, JSON‑based log of HTTP requests and responses that is used primarily for network-level debugging and performance analysis. 

> 5. Click **Done**.

Reviewer: ```suggestion
The file will start downloading
```

^ Just a suggestion I typically favor not having a "Click done" or "click save" step and letting the UI manage that portion if I can help it.

### Special resource IDs
To ensure workspace identity is preserved at import and avoid collisions, exported Insomnia files include special resource IDs like `__WORKSPACE_ID__` and `__BASE_ENVIRONMENT_ID__`. Insomnia automatically replaces these during import to reconstruct the structure accurately.

For more information, go to our [Special resource IDs](/insomnia/special-resource-ids/) reference page.

### Resource types
The export file includes a `data.resources` object with a `resources` array, each element represents an object of a specific resource type:
- `_type` field indicates the type of resource: `workspace`, `environment`, `request`, `response`, `folder`, `mock`, `plugin`, and `test`.
- Only supported resource types that are defined in Insomnia’s schema are included; internal metadata or auxiliary data is excluded.
These resource types determine what you create when you import. If a resource type is omitted or unsupported, the related element won't appear in the reconstructed workspace.

> For more information, go to our [Resource types reference](/insomnia/resource-types-reference/) reference page.

Reviewer: How-to's should be goal oriented in almost all cases. Adding this info at the bottom of the doc breaks away from the how-to framework. 

I think you can delete this, and from within the steps, link to: 
* [Resource types reference](/insomnia/resource-types-reference/) reference page.
* For more information, go to our [Special resource IDs](/insomnia/special-resource-ids/) reference page.

> ### Special resource IDs
> To ensure workspace identity is preserved at import and avoid collisions, exported Insomnia files include special resource IDs like `__WORKSPACE_ID__` and `__BASE_ENVIRONMENT_ID__`. Insomnia automatically replaces these during import to reconstruct the structure accurately.
> 
> For more information, go to our [Special resource IDs](/insomnia/special-resource-ids/) reference page.
> 
> ### Resource types
> The export file includes a `data.resources` object with a `resources` array, each element represents an object of a specific resource type:
> - `_type` field indicates the type of resource: `workspace`, `environment`, `request`, `response`, `folder`, `mock`, `plugin`, and `test`.
> - Only supported resource types that are defined in Insomnia’s schema are included; internal metadata or auxiliary data is excluded.
> These resource types determine what you create when you import. If a resource type is omitted or unsupported, the related element won't appear in the reconstructed workspace.
> 
> For more information, go to our [Resource types reference](/insomnia/resource-types-reference/) reference page.

Reviewer: ```suggestion
```
